nemo/collections/asr/parts/utils/longform_clustering.py

```diff
@@ -0,0 +1,449 @@
+                window_range_list.append([window_offset, window_offset + merged_embs.shape[0]])
+                window_offset += merged_embs.shape[0]
+
+        if not torch.jit.is_scripting():
+            pbar.close()
+
+        # Concatenate the reduced embeddings then perform high-level clustering
+        reduced_embs = torch.cat(total_emb)
+        reduced_mat = getCosAffinityMatrix(reduced_embs)
+
+        # Step-4: Map the aggregated labels `Y_aggr` back to the original labels for all `org_len` input embeddings: `Y_unpack`
+        Y_aggr = self.speaker_clustering.forward_unit_infer(
+            mat=reduced_mat,
+            oracle_num_speakers=oracle_num_speakers,
+            max_rp_threshold=max_rp_threshold,
+            max_num_speakers=max_num_speakers,
+            sparse_search_volume=sparse_search_volume,
+            fixed_thres=fixed_thres,
+        )
+        if reduced_embs.shape[0] != Y_aggr.shape[0]:
+            raise ValueError(
+                f"The number of embeddings ({reduced_embs.shape[0]}) and the number of clustered labels ({Y_aggr.shape[0]}) do not match."
+            )
+
+        # Reassign the labels to the original embeddings
+        Y_unpack = self.unpack_labels(
+            Y_aggr=Y_aggr,
+            window_range_list=window_range_list,
+            absolute_merge_mapping=absolute_merge_mapping,
+            org_len=emb.shape[0],
+        )
+        if Y_unpack.shape[0] != emb.shape[0]:
+            raise ValueError(
+                f"The number of raw input embeddings ({emb.shape[0]}) and the number of clustered labels ({Y_unpack.shape[0]}) do not match."
+            )
+        return Y_unpack
+
+    def forward_infer(
+        self,
+        embeddings_in_scales: torch.Tensor,
+        timestamps_in_scales: torch.Tensor,
+        multiscale_segment_counts: torch.LongTensor,
+        multiscale_weights: torch.Tensor,
+        oracle_num_speakers: int = -1,
+        max_rp_threshold: float = 0.15,
+        max_num_speakers: int = 8,
+        enhanced_count_thres: int = 80,
+        sparse_search_volume: int = 30,
+        fixed_thres: float = -1.0,
+        chunk_cluster_count: int = 50,
+        embeddings_per_chunk: int = 10000,
+    ) -> torch.LongTensor:
+        """
+        This function is a wrapper designed for toggling between long-form and short-form speaker clustering.
+        The details of short-form clustering is in `SpeakerClustering` class.
+        NOTE: `torch.jit.script` currently does not support `**kwargs` in the function signature therefore,
+        we need to use a wrapper function to handle the arguments.
+        """
+        self.embeddings_per_chunk = (
+            embeddings_per_chunk if embeddings_per_chunk is not None else self.embeddings_per_chunk
+        )
+        self.chunk_cluster_count = chunk_cluster_count if chunk_cluster_count is not None else self.chunk_cluster_count
+        if embeddings_in_scales.shape[0] > self.embeddings_per_chunk:
```

Comment:
What you pointed out is correct. Therefore, I changed the line to use `torch.max(multiscale_segment_counts)` so that it checks out the number of embedding vectors that is being clustered.